### PR TITLE
debug: extract & train code validated (with issue)

### PR DESCRIPTION
**Pull request by jun-brro**
class update
reshape logic update
activation extraction logic updated
SAE triain confirmed

TODO: SAE overfitting

### CODE REVIEW

**Comment by jun-brro** [comment]
2025-10-04 20:21:16,853 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 17 train_loss=0.133936 val_
loss=1577.912842                                                                                    
2025-10-04 20:21:16,869 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 18 train_loss=0.118242 val_
loss=1569.378174                                                                                    
2025-10-04 20:21:16,885 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 19 train_loss=0.106030 val_
loss=1530.034424                                                                                    
2025-10-04 20:21:17,029 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 20 train_loss=0.096659 val_
loss=1499.554932                                                                                    
2025-10-04 20:21:17,170 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 21 train_loss=0.089712 val_
loss=1457.022583                                                                                    
2025-10-04 20:21:17,312 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 22 train_loss=0.083745 val_
loss=1422.177979                                                                                    
2025-10-04 20:21:17,451 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 23 train_loss=0.078944 val_
loss=1376.942383                                                                                    
2025-10-04 20:21:17,593 INFO [vlm_intrinsic_tox.sae.train] Layer 0 epoch 24 train_loss=0.073089 val_
loss=1332.889282    


overfitting occurs!!

**Comment by h-albert-lee** [comment]
Confirmed

**Comment by jun-brro** [comment]
update: train/val loss scale unify -> 

INFO [vlm_intrinsic_tox.sae.train] Layer 4 epoch 24 train_loss=0.053957 val_loss=0.433179

**Comment by jun-brro** [comment]
import error updated.

TODO:
- select ex-toxic/benign pair data (or generate)
- label system (benign: 0, ex-toxic: 1, in-toxic: 2)
- eval system update (sync with label system)

Train object - acquire less than 0.2 in test loss (w/ more than 10k trainset total)

**Comment by h-albert-lee** [comment]
@jun-brro 일단 한번 Merge할까요?

**Comment by jun-brro** [comment]
좋습니다!
